src/module/item/affliction/data.ts

```diff
@@ -1,90 +1,383 @@
+                        token: new fields.DocumentUUIDField({ required: false, nullable: true, initial: null }),
+                    },
+                    { required: false, nullable: true, initial: null },
+                ),
+                roll: new fields.SchemaField(
+                    {
+                        total: new fields.NumberField({ required: true }),
+                        degreeOfSuccess: new fields.NumberField<
+                            DegreeOfSuccessIndex,
+                            DegreeOfSuccessIndex,
+                            false,
+                            true,
+                            true
+                        >({
+                            required: false,
+                            nullable: true,
+                            initial: null,
+                            min: 0,
+                            max: 3,
+                        }),
+                    },
+                    { required: true, nullable: true, initial: null },
+                ),
+            },
+            { required: false, nullable: true, initial: null },
+        );
+    }
 }
 
+type EffectContextDataSchema = {
+    origin: fields.SchemaField<{
+        actor: fields.DocumentUUIDField<ActorUUID, true, false>;
+        token: fields.DocumentUUIDField<TokenDocumentUUID, false, true, true>;
+        item: fields.DocumentUUIDField<ItemUUID, false, true, true>;
+        spellcasting: fields.SchemaField<
+            EffectContextSpellcastingSchema,
+            SourceFromSchema<EffectContextSpellcastingSchema>,
+            ModelPropsFromSchema<EffectContextSpellcastingSchema>,
+            false,
+            true,
+            true
+        >;
+        rollOptions: fields.ArrayField<fields.StringField, string[], string[], false, false>;
+    }>;
+    target: fields.SchemaField<
+        EffectContextTargetSchema,
+        SourceFromSchema<EffectContextTargetSchema>,
+        ModelPropsFromSchema<EffectContextTargetSchema>,
+        false,
+        true,
+        true
+    >;
+    roll: fields.SchemaField<
+        EffectContextRollSchema,
+        SourceFromSchema<EffectContextRollSchema>,
+        ModelPropsFromSchema<EffectContextRollSchema>,
+        true,
+        true
+    >;
+};
+
+type EffectContextSpellcastingSchema = {
+    attribute: fields.SchemaField<{
+        type: fields.StringField<AttributeString, AttributeString, true, false>;
+        mod: fields.NumberField<number, number, true, false>;
+    }>;
+    tradition: fields.StringField<MagicTradition, MagicTradition, false, true, true>;
+};
+
+type EffectContextTargetSchema = {
+    actor: fields.DocumentUUIDField<ActorUUID, true, false>;
+    token: fields.DocumentUUIDField<TokenDocumentUUID, false, true, true>;
+};
+
+type EffectContextRollSchema = {
+    degreeOfSuccess: fields.NumberField<DegreeOfSuccessIndex, DegreeOfSuccessIndex, false, true, true>;
+    total: fields.NumberField<number, number, true, false>;
```

Comment:
```suggestion
    total: fields.NumberField<number, number, true, false, true>;
```

There's a mistaken in `NumberField`: only if it is _nullable_ is there a default initial value.

Reply:
In v13-dev, setting `required` on a `Numberfield` to true has `initial` defaulting to have a value. Since it's omitted here, I assumed that was your intent. But if the opposite is your intent, fill it in that way.

Reply:
The opposite was my intent, but your suggestion had set initial set to true when your words seemed to imply otherwise.
I changed it already though.